llm/app/services/visitors_service.py

```
@@ -24,897 +24,585 @@ def __init__(self):
-                pass
-
-            logger.info(f"✅ 메모 업데이트 성공: {client_code}")
-            return True
-
+                raise APIException("서버 내부 오류가 발생했습니다", 500, api_name)
+            else:
+                raise APIException(f"API 호출 실패: HTTP {response.status_code}", response.status_code, api_name)
+                
         except requests.exceptions.Timeout:
-            raise APIException("서버 응답 시간이 초과되었습니다", None, "update_memo")
+            raise APIException("서버 응답 시간이 초과되었습니다", None, api_name)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)